build.gradle

```diff
@@ -1,19 +1,60 @@
 plugins {
     id 'java'
+    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'org.springframework.boot' version '2.4.3'
+
 }
 
-group 'org.example'
-version '1.0-SNAPSHOT'
+subprojects {
+    apply plugin: 'org.springframework.boot'
+    apply plugin: 'io.spring.dependency-management'
+    apply plugin: 'java-library'
 
-repositories {
-    mavenCentral()
-}
+    group = 'com.epam.esm'
+    version = '1.0.0'
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
 
-dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
-}
+    configurations {
+        compileOnly {
+            extendsFrom annotationProcessor
+        }
+    }
+
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
```

Comment:
Try to distribute dependencies among the appropriate modules.

Reply:
Specific dependencies (which were only for one module) were distributed but the dependencies  which were common for all modules were abandoned in main build.gradle